packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -0,0 +1,500 @@
+            );
+          }
+        )
+      ),
+    );
+  }
+
+  static Widget _delegatedCoverSheetSecondaryTransition(Animation<double> secondaryAnimation, Widget? child) {
+    const Curve curve = Curves.linearToEaseOut;
+    const Curve reverseCurve = Curves.easeInToLinear;
+    final CurvedAnimation curvedAnimation = CurvedAnimation(
+      curve: curve,
+      reverseCurve: reverseCurve,
+      parent: secondaryAnimation
+    );
+
+    final Animation<Offset> slideAnimation = curvedAnimation.drive(_kMidUpTween);
+    final Animation<double> scaleAnimation = curvedAnimation.drive(_kScaleTween);
+    curvedAnimation.dispose();
+
+
+    return SlideTransition(
+      position: slideAnimation,
+      transformHitTests: false,
+      child: ScaleTransition(
+        scale: scaleAnimation,
+        filterQuality: FilterQuality.medium,
+        alignment: Alignment.topCenter,
+        child: ClipRRect(
+          borderRadius: const BorderRadius.vertical(top: Radius.circular(12)),
+          child: child,
+        ),
+      ),
+    );
+  }
+
+  @override
+  State<CupertinoSheetTransition> createState() => _CupertinoSheetTransitionState();
+}
+
+class _CupertinoSheetTransitionState extends State<CupertinoSheetTransition> {
+
+  // When this page is coming in to cover a non-sheet page.
+  late Animation<Offset> _primaryPositionAnimation;
+  // When this page is coming in to cover another sheet.
+  late Animation<Offset> _primarySheetPositionAnimation;
+  // The offset animation when this page is becoming covered by another sheet.
+  late Animation<Offset> _secondaryPositionAnimation;
+  // The scale animation when this page is becoming covered by another sheet.
+  late Animation<double> _secondaryScaleAnimation;
+  // Curve of primary page which is coming in to cover another route.
+  CurvedAnimation? _primaryPositionCurve;
+  // Curve of secondary page which is becoming covered by another sheet.
+  CurvedAnimation? _secondaryPositionCurve;
+
+  @override
+  void initState() {
+    super.initState();
+    _setupAnimation();
+  }
+
+  @override
+  void didUpdateWidget(covariant CupertinoSheetTransition oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.primaryRouteAnimation != widget.primaryRouteAnimation
+    || oldWidget.secondaryRouteAnimation != widget.secondaryRouteAnimation) {
+      _disposeCurve();
+      _setupAnimation();
+    }
+  }
+
+  @override
+  void dispose() {
+    _disposeCurve();
+    super.dispose();
+  }
+
+  void _setupAnimation() {
+    _primaryPositionCurve = CurvedAnimation(
+      curve: Curves.fastEaseInToSlowEaseOut,
+      reverseCurve: Curves.fastEaseInToSlowEaseOut.flipped,
+      parent: widget.primaryRouteAnimation,
+    );
+    _secondaryPositionCurve = CurvedAnimation(
+      curve: Curves.linearToEaseOut,
+      reverseCurve: Curves.easeInToLinear,
+      parent: widget.secondaryRouteAnimation
+    );
+     _primaryPositionAnimation = (_primaryPositionCurve ?? widget.primaryRouteAnimation)
```

Comment:
same for other property below

Reply:
The CurvedAnimations are disposed of on widget dispose and update, so this is a failsafe in case there's a spot were it is null and not set. Theoretically that shouldn't happen. I copied the implementation for Cupertino route https://github.com/flutter/flutter/blob/885a5ca265cb74355d3741da249c3006f58a2427/packages/flutter/lib/src/cupertino/route.dart#L528

Reply:
> his is a failsafe in case there's a spot were it is null and not set.

but we just set it in line 283, I can't think of a code path this can happen